sdk/samples/iot/iot_sample_common.c

```diff
@@ -496,8 +496,9 @@ void iot_sample_generate_sas_base64_encoded_signed_signature(
   // Decode the sas base64 encoded key to use for HMAC signing.
   char sas_decoded_key_buffer[64];
   az_span sas_decoded_key = AZ_SPAN_FROM_BUFFER(sas_decoded_key_buffer);
-  if (az_result_failed(
-          rc = decode_base64_bytes(sas_base64_encoded_key, sas_decoded_key, &sas_decoded_key)))
+
+  rc = decode_base64_bytes(sas_base64_encoded_key, sas_decoded_key, &sas_decoded_key);

```

Comment:
nit consider declaring it here?
```suggestion
  int rc = decode_base64_bytes(sas_base64_encoded_key, sas_decoded_key, &sas_decoded_key);
```

Reply:
good eye

Reply:
I want to keep it at the top unintialized because it is initialized/change multiple times through the function, rather than a one time use.